k8s/operators/src/diskpool/main.rs

```diff
@@ -619,20 +645,19 @@ impl ResourceContext {
 /// is wrong would be to consult the logs.
 async fn ensure_crd(k8s: Client) {
     let dsp: Api<k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition> = Api::all(k8s);
-    let lp = ListParams::default().fields(&format!("metadata.name={}", "diskpools.openebs.io"));
-    let crds = dsp.list(&lp).await.expect("failed to list CRDS");
 
-    // the CRD has not been installed yet, to avoid overwriting (and create upgrade issues) only
-    // install it when there is no crd with the given name
-    if crds.iter().count() == 0 {
-        let crd = DiskPool::crd();
+    // Check if there is an existing dsp crd. If yes, Replace it with new generated one.
+    // Create new if it doesnt exist.
+    let mut crd = DiskPool::crd();
+    if let Ok(exiting) = dsp.get("diskpools.openebs.io").await {
+        crd.metadata.resource_version = exiting.resource_version();
```

Comment:
Should we only patch if version differs?

Reply:
resource_version is not present in auto generated crd. it is added by k8s once object is updated. How would we know if version is different?

Reply:
Not this one, we should bump the CRD api version from v1alpha1 to something higher. Then we can check this, and if != then update.

Reply:
Could be tedious to update crd resource version manually for any changes in crd. If we use this validation we have to bump it everytime right?  With current code it just replaces with crd from the operator. If someone doesnt update crd version new changes will not be patched and new field will not reflect in CR. So thought of just replacing it regardless. Should not impact if there is no change in crd.

Reply:
If it's tedious then we automate it, but we should update the version, otherwise what's the point of having a version in the CRD?
I'm fine with updating everytime then, but let's update the api version anyway.

Reply:
Okay will add it.